src/MCPClient/lib/clientScripts/verify_checksum.py

```diff
@@ -0,0 +1,276 @@
+        """Return version information for the command being called."""
+        try:
+            return self._call("--version")[0]
+        except subprocess.CalledProcessError:
+            return self.COMMAND
+
+    def get_command_detail(self):
+        """Provide some way for the user to get information out of the class to
+        write METS/PREMIS provenance information.
+        """
+        if not self.command_called:
+            raise PREMISFailure()
+        return 'program="{}"; version="{}"'.format(
+            " ".join(self.command_called), self.version()
+        )
+
+    @staticmethod
+    def _decode(out):
+        """Decode string output in Py2 or Py3 and return a list of lines to be
+        parsed elsewhere.
+        """
+        try:
+            return str(out, "utf8").split("\n")
+        except TypeError:
+            return out.decode("utf8").split("\n")
+
+    @staticmethod
+    def get_ext(path):
+        """Return the extension of the checksum file provided"""
+        ext = os.path.splitext(path)[1]
+        if not ext:
+            return path
+        return ext.replace(".", "")
+
+    @staticmethod
+    def _count_lines(path):
+        """Count the number of lines in a checksum file."""
+        count = 0
+        with open(path) as hashfile:
+            for count, _ in enumerate(hashfile):
+                pass
+        # Negate zero-based count.
+        return count + 1
+
+    @staticmethod
+    def _count_files(path):
+        """Walk the directories on a given path and count the number of files.
+        """
+        return sum([len(files) for _, _, files in os.walk(path)])
+
+
+def write_premis_event_per_file(transfer_uuid, event_detail):
+    """Write a PREMIS event for every file in the transfer on success."""
+    group_type = "transfer_id"
+    event_type = "fixity_check"
+    event_outcome = "pass"
+    # Retrieve file UUIDs from the database.
+    kwargs = {"removedtime__isnull": True, group_type: transfer_uuid}
+    with transaction.atomic():
+        file_uuids = File.objects.filter(**kwargs).values_list("uuid")
+        if not file_uuids:
+            raise PREMISFailure()
+        for (fileUUID,) in file_uuids:
+            insertIntoEvents(
+                fileUUID=fileUUID,
+                eventType=event_type,
+                eventDateTime=datetime.datetime.now()
+                .replace(microsecond=0)
+                .isoformat(),
+                eventDetail=event_detail,
+                eventOutcome=event_outcome,
+                eventOutcomeDetailNote="",
+            )
+
+
+def run_hashsum_commands(job):
+    """Run hashsum commands and generate a cumulative return code."""
+    transfer_dir = None
+    transfer_uuid = None
+    try:
+        transfer_dir = job.args[1]
+        transfer_uuid = job.args[2]
+    except IndexError:
+        logger.error("Cannot access expected module arguments: %s", job.args)
+        return 1
+    ret = 0
+    for hashfile in Hashsum.COMMANDS:
```

Comment:
Can't think of a better name alternative for `COMMANDS` but this iteration seems "off?" (sorry can't think of a better word either!), like "shouldn't this be iterating through commands?".

Reply:
I see what you mean. I wrestled with "COMMANDS" "ALGORITHMS" but I have settled on "HASHFILES" with an additional comment associated with the data structure it is referring to. Will that work do you think?